File: src/main/java/k31/grc/ast/node/cond/CondRelOpBase.java
package k31.grc.ast.node.cond;

import k31.grc.ast.node.expr.ExprBase;
import k31.grc.ast.visitor.Visitor;

public abstract class CondRelOpBase extends CondBase {

	private ExprBase left;
	private ExprBase right;

	public ExprBase getLeft() {
		return left;
	}

	public ExprBase getRight() {
		return right;
	}

	public void setLeft(ExprBase left) {
		this.left = left;
	}

	public void setRight(ExprBase right) {
		this.right = right;
	}

	public CondRelOpBase(String text) {
		super(text);
	}

	@Override
	public void accept(Visitor v) {

		v.visit(this);
	}
}

File: DotNetGrc/Grc/Ast/Node/Cond/CondRelOpBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Grc.Ast.Node.Expr;
using Grc.Ast.Visitor;

namespace Grc.Ast.Node.Cond
{
	public abstract partial class CondRelOpBase : CondBase
	{
		private readonly ExprBase left;
		private readonly ExprBase right;

		private readonly string oper;

		public ExprBase Left { get { return left; } }

		public ExprBase Right { get { return right; } }

		public override int Line { get { return left.Line; } }

		public override int Pos { get { return left.Pos; } }

		protected CondRelOpBase(ExprBase left, ExprBase right, string oper)
		{
			this.left = left;
			this.right = right;

			this.oper = oper;
		}

		public override void Accept(IVisitor v)
		{
			v.Visit(this);
		}

		public override string Text
		{
			get { return string.Format("({0} {1} {2})", left.Text, oper, right.Text); }
		}

		public override string ToString()
		{
			return oper;
		}
	}
}
